pycqed/measurement/sweep_points.py

```diff
@@ -270,44 +270,59 @@ def get_meas_obj_sweep_points_map(self, measurement_objects):
 
         If the sweep dimension has more than one sweep parameter name (dict with
         several keys), then:
-            - first tries to add to the list for each mobj only those sweep
+            - first adds to the list for each mobj only those sweep
             param names that contain the mobj_name.
-            - If it can't find the mobj_name in the sweep param name, assumes
-            there is only one param per mobj in each sweep dimension, and that
-            the order of params in each sweep dimension corresponds to the
-            order of keys in keys_list.
-            I.e. key_i in sweep_points[0] contains the sweep information for
-            measured_objects[i].
-
-        :param measured_objects: list of strings to be used as keys in the
-            returned dictionary. These are the measured object names
+            - if some parameters remain that do not contain any of the
+            measurement_objects names, it is assumed that all
+            measurement_objects used them so they will be added for each mobj
+
+        :param measurement_objects: list of strings to be used as keys in the
+            returned dictionary. These are the measured object names.
+            Can also be list of QuDev_transmon instanced in which case
+            this function gets the list of names.
         :return: dict of the form
          {mobj_name: [sweep_param_name_0, ..., sweep_param_name_n]}
         """
+
+        # Ensure measurement_objects is a list
         if isinstance(measurement_objects, list):
             measurement_objects = copy(measurement_objects)
         else:
             measurement_objects = [measurement_objects]
+
         for i, mobj in enumerate(measurement_objects):
             if hasattr(mobj, 'name'):
+                # list of QuDev_transmon instances was provided
```

Comment:
Same as above: the comment can be written in a more general way: in the future, we might have other mobjs that have a name attribute.

Reply:
Changed now